internal/cmd/plugin/destroy/destroy.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package destroy implements a command to destroy an instances of a cluster and its associated PVC
+package destroy
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin"
+)
+
+// Destroy implements the destroy subcommand
+func Destroy(ctx context.Context, clusterName, instanceID string, keepPVC bool) error {
+	// Check if the Pod exist
+	var pod v1.Pod
+	instanceName := fmt.Sprintf("%s-%s", clusterName, instanceID)
+
+	err := plugin.Client.Get(ctx, client.ObjectKey{
+		Namespace: plugin.Namespace,
+		Name:      instanceName,
+	}, &pod)
+
+	var volumes []v1.Volume
+	if err != nil {
```

Comment:
we should not rely on pod existing, this could lead to inconsistencies if any error is encountered after this step

Reply:
What would the functionality be if a pod did not exists? There is no reason to re-implement kubectl delete pvc. I could see maybe a flag to clean all of the pvc's where the status is set to 'detached' but other than that I think if there is no pod or errors then it is time to use kubectl and not the cnpg plugin.

Reply:
passing this to @mnencia to have further clarification on behavior

Reply:
Okay so when I add the 'detached' annotation I am now adding the instanceName label (which will be already done in version 1.16+). Then if the command is invoked without the --keep-pvc flag and no pod is found it will still check for pvc's with the instanceName label and remove all that match.